quickwit/quickwit-cli/tests/cli.rs

```diff
@@ -866,89 +869,85 @@ async fn test_all_local_index() -> Result<()> {
 #[tokio::test]
 #[serial]
 #[cfg_attr(not(feature = "ci-test"), ignore)]
-async fn test_cmd_all_with_s3_localstack_cli() -> Result<()> {
+async fn test_all_with_s3_localstack_cli() {
     let index_id = append_random_suffix("test-all--cli-s3-localstack");
-    let test_env = create_test_env(index_id, TestStorageType::S3)?;
-    make_command(
-        format!(
-            "index create --index-config {} --config {}",
-            test_env.resource_files["index_config"].display(),
-            test_env.resource_files["config"].display()
-        )
-        .as_str(),
-    )
-    .assert()
-    .success();
-
-    test_env
-        .metastore()
-        .await?
-        .index_metadata(&test_env.index_id)
-        .await
-        .unwrap();
+    let test_env = create_test_env(index_id.clone(), TestStorageType::S3).unwrap();
+    create_logs_index(&test_env);
 
     ingest_docs(test_env.resource_files["logs"].as_path(), &test_env);
 
-    // cli search
-    make_command(
-        format!(
-            "index search --index {} --query level:info --config {}",
-            test_env.index_id,
-            test_env.resource_files["config"].display(),
-        )
-        .as_str(),
-    )
-    .assert()
-    .success()
-    .stdout(predicate::function(|output: &[u8]| {
-        let result: Value = serde_json::from_slice(output).unwrap();
-        result["num_hits"] == Value::Number(Number::from(2i64))
-    }));
+    // Cli search
+    let args = SearchIndexArgs {
+        config_uri: test_env.config_uri.clone(),
+        index_id: index_id.clone(),
+        query: "level:info".to_string(),
+        aggregation: None,
+        max_hits: 20,
+        start_offset: 0,
+        search_fields: None,
+        snippet_fields: None,
+        start_timestamp: None,
+        end_timestamp: None,
+        data_dir: None,
+        sort_by_score: false,
+    };
+
+    let search_res = search_index(args).await.unwrap();
+    assert_eq!(search_res.num_hits, 2);
 
     // serve & api-search
     // TODO: ditto.
-    let mut server_process = spawn_command(
-        format!(
-            "run --service searcher --service metastore --config {}",
-            test_env.resource_files["config"].display(),
-        )
-        .as_str(),
-    )
-    .unwrap();
+    let run_cli_command = RunCliCommand {
+        config_uri: test_env.config_uri.clone(),
+        data_dir_path: None,
+        services: Some(HashSet::from([
+            QuickwitService::Searcher,
+            QuickwitService::Metastore,
+        ])),
+        metastore_uri: None,
+        cluster_id: None,
+        node_id: None,
+        peer_seeds: None,
+    };
+
+    let service_task = tokio::spawn(async move { run_cli_command.execute().await.unwrap() });
+
     // TODO: ditto.
     sleep(Duration::from_secs(2)).await;
```

Comment:
Can you try to remove that sleep by polling some health endpoint until the searcher is ready?